tests/test_js_graph.py

```diff
@@ -104,6 +104,22 @@ def test_insert_edge(graph_cases, mode):
     assert graph.get_edge(*graph.get_vertices()) is not None
 
 
+def test_get_terminal_points(graph_cases):
+    """
+    :type graph_cases: qmxgraph.tests.conftest.GraphCaseFactory
+    """
+    graph = graph_cases('2v_1e')
+    edge_id = graph.get_id(graph.get_edge(*graph.get_vertices()))
+    terminal_points = graph.eval_js_function("api.getEdgeTerminalPoints", edge_id)
+    (source_x, source_y), (target_x, target_y) = terminal_points
+
+    from numbers import Number
+    assert isinstance(source_x, Number)
```

Comment:
Nit: shouldn't this always be a `float`? I mean, if an `int` or `complex` is returned here, we probably have a bug in the code.

Reply:
Maybe. I can change this to reflect that.
I will use the effective numbers since they shouldn't realy change.

Reply:
OK, sounds good. If using the real numbers proves to be a pain, we can change to `assert isinstance(source_x, float)` later.